### PR TITLE
[HOPSWORKS-1992] add a gemserver

### PR DESCRIPTION
**Pull request by jimdowling**
https://logicalclocks.atlassian.net/browse/HOPSWORKS-1992

### CODE REVIEW

File: recipes/gemserver.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    user "root"
+    code <<-EOH
+        cd /tmp
+        yum install wget -y
+        wget -nc #{node['download_url']}/#{node['setup']['chefdk']['rhel']}
+        yum install -y #{node['setup']['chefdk']['rhel']} 
+      EOH
+  end
+
+end
```
Comment (SirOibaf) [inline recipes/gemserver.rb:25]: This part doesn't make much sense. We run setup-chef with Karamel, which as first step already downloads and install chefdk. So by the time we get here, chefdk is already installed. 

Additionally yum will fail if the package you are trying to install already exists. So this whole recipe is going to fail

File: attributes/default.rb [hunk 2]
```diff
@@ -44,3 +44,8 @@
 
 default['setup']['disable_nw_mgr']                 = "false"
 
+default['setup']['chefdk']['rhel']                 = "chefdk-3.7.23-1.el7.x86_64.rpm
+default['setup']['chefdk']['debian']               = "chefdk_3.7.23-1_amd64.deb"
+
+default['setup']['gemserver']['version']           = "2.5.0"
```
Comment (SirOibaf) [inline attributes/default.rb:50]: As I mentioned in the Karamel PR, I'm quite confused on why you used 2.5.0. On all my VMs the embedded version is 2.4.0

File: recipes/gemserver.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+bash "start-gem-server" do
+  user "root"
+  code <<-EOH
+      set -e
+      if [ ! -d /opt/chefdk/embedded/lib/ruby/gems/#{node['setup']['gemserver']['version']} ] ; then
+         echo "Wrong gem server version. Set the correct attribute value for: setup/gemserver/version" 
+         echo "Check in "ls /opt/chefdk/embedded/lib/ruby/gems/"
+         exit 5
+      fi
+      nohup /opt/chefdk/embedded/bin/gem server --dir --port #{node['setup']['gemserver']['port']} /opt/chefdk/embedded/lib/ruby/gems/#{node['setup']['gemserver']['version']}/gems &
```
Comment (SirOibaf) [inline recipes/gemserver.rb:37]: I'd make the path configurable. As I mentioned in the Karamel's PR, you need more gems than what the embedded ruby version provides.

**Review by SirOibaf — CHANGES_REQUESTED** [review]

File: recipes/gemserver.rb (hunk 1 above)
Comment (jimdowling) [reply recipes/gemserver.rb:25]: This didn't work. It's been done in karamel. Closing.

**Pull request closed without merging**